Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -1022,6 +1033,669 @@ void vtkMRMLSliceLogic::UpdateReconstructionSlab(vtkMRMLSliceLogic* sliceLogic,
+  }
+
+  vtkSmartPointer<vtkImageData> gridImage = vtkSmartPointer<vtkImageData>::New();
+  gridImage->SetOrigin(transformGridOrigin);
+  gridImage->SetDimensions(gridDimensions);
+  gridImage->SetSpacing(gridSpacing);
+  gridImage->AllocateScalars(VTK_DOUBLE, 3);
+  vtkSmartPointer<vtkOrientedGridTransform> transform = vtkSmartPointer<vtkOrientedGridTransform>::New();
+  transform->SetDisplacementGridData(gridImage);
+  transform->SetGridDirectionMatrix(newGridDirectionMatrix);
+  transformToStraightenedNode->SetAndObserveTransformFromParent(transform);
+
+  vtkSmartPointer<vtkAppendPolyData> appender;
+  if (reslicingPlanesModelNode != nullptr)
+  {
+    appender = vtkSmartPointer<vtkAppendPolyData>::New();
+  }
+
+  // Currently there is no API to set PreferredInitialNormalVector in the curve
+  // coordinate system, therefore a new coordinate system generator must be set up:
+  vtkSmartPointer<vtkParallelTransportFrame> curveCoordinateSystemGeneratorWorld =
+    vtkSmartPointer<vtkParallelTransportFrame>::New();
+  curveCoordinateSystemGeneratorWorld->SetInputData(resampledCurveNode->GetCurveWorld());
+  curveCoordinateSystemGeneratorWorld->SetPreferredInitialNormalVector(transformGridAxisX);
+  curveCoordinateSystemGeneratorWorld->Update();
+  vtkPolyData* curvePoly = curveCoordinateSystemGeneratorWorld->GetOutput();
+  vtkPointData* pointData = curvePoly->GetPointData();
+  vtkDoubleArray* normals = vtkDoubleArray::SafeDownCast(
+    pointData->GetAbstractArray(curveCoordinateSystemGeneratorWorld->GetNormalsArrayName()));
+  vtkDoubleArray* binormals = vtkDoubleArray::SafeDownCast(
+    pointData->GetAbstractArray(curveCoordinateSystemGeneratorWorld->GetBinormalsArrayName()));
+
+  // Compute displacements
+  vtkGridTransform* transformGrid = vtkGridTransform::SafeDownCast(transformToStraightenedNode->GetTransformFromParent());
+  vtkImageData* displacementGrid = transformGrid->GetDisplacementGrid();
+  vtkDataArray* transformDisplacements_RAS = displacementGrid->GetPointData()->GetScalars();
+
+  for (int gridK = 0; gridK < gridDimensions[2]; ++gridK)
+  {
+    // The curve's built-in coordinate system generator could be used like this
+    // (if it had PreferredInitialNormalVector exposed):
+    //
+    // curvePointToWorld = vtk.vtkMatrix4x4()
+    // resampledCurveNode.GetCurvePointToWorldTransformAtPointIndex(
+    //     resampledCurveNode.GetCurvePointIndexFromControlPointIndex(gridK),
+    //     curvePointToWorld,
+    // )
+    // curvePointToWorldArray = slicer.util.arrayFromVTKMatrix(curvePointToWorld)
+    // curveAxisX_RAS = curvePointToWorldArray[0:3, 0]
+    // curveAxisY_RAS = curvePointToWorldArray[0:3, 1]
+    // curvePoint_RAS = curvePointToWorldArray[0:3, 3]
+    //
+    // But now we get the values from our own coordinate system generator:
+    const int curvePointIndex = resampledCurveNode->GetCurvePointIndexFromControlPointIndex(gridK);
+    const double* curveAxisX_RASVec = normals->GetTuple3(curvePointIndex);
```

Comment:
```suggestion
    const double* curveAxisX_RAS = normals->GetTuple3(curvePointIndex);
```

Since by definition an axis is a directed line, I suggest to remove the suffix `Vec`. 

Make sure to update the remaining of the contributed code to match this convention :pray: